tests/foreman/ui/test_registration.py

```diff
@@ -0,0 +1,337 @@
+
+    :parametrized: yes
+
+    :CaseLevel: Integration
+    """
+    # make sure global parameters for rex and insights are set to true
+    insights_cp = (
+        target_sat.api.CommonParameter()
+        .search(query={'search': 'name=host_registration_insights'})[0]
+        .read()
+    )
+    rex_cp = (
+        target_sat.api.CommonParameter()
+        .search(query={'search': 'name=host_registration_remote_execution'})[0]
+        .read()
+    )
+
+    if not insights_cp.value:
+        target_sat.api.CommonParameter(id=insights_cp.id, value=1).update(['value'])
+    if not rex_cp.value:
+        target_sat.api.CommonParameter(id=rex_cp.id, value=1).update(['value'])
+
+    # rex interface
+    iface = 'eth0'
+    # fill in the global registration form
+    with session:
+        cmd = session.host.get_register_command(
+            {
+                'general.organization': module_org.name,
+                'general.location': smart_proxy_location.name,
+                'general.operating_system': default_os.title,
+                'general.activation_keys': module_activation_key.name,
+                'advanced.update_packages': True,
+                'advanced.rex_interface': iface,
+                'general.insecure': True,
+            }
+        )
+    expected_pairs = [
+        f'organization_id={module_org.id}',
+        f'activation_keys={module_activation_key.name}',
+        f'location_id={smart_proxy_location.id}',
+        f'operatingsystem_id={default_os.id}',
+        f'{default_smart_proxy.name}',
+        'insecure',
+        'update_packages=true',
+    ]
+    for pair in expected_pairs:
+        assert pair in cmd
+    # rhel repo required for insights client installation,
+    # syncing it to the satellite would take too long
+    rhelver = rhel_contenthost.os_version.major
+    if rhelver > 7:
+        rhel_contenthost.create_custom_repos(**settings.repos[f'rhel{rhelver}_os'])
+    else:
+        rhel_contenthost.create_custom_repos(
+            **{f'rhel{rhelver}_os': settings.repos[f'rhel{rhelver}_os']}
+        )
+    # make sure there will be package availabe for update
+    if rhel_contenthost.os_version.major == '6':
+        package = FAKE_1_CUSTOM_PACKAGE
+        repo_url = settings.repos.yum_1['url']
+    else:
+        package = FAKE_7_CUSTOM_PACKAGE
+        repo_url = settings.repos.yum_3['url']
+    rhel_contenthost.create_custom_repos(fake_yum=repo_url)
+    rhel_contenthost.execute(f"yum install -y {package}")
+    # run curl
+    result = rhel_contenthost.execute(cmd)
+    assert result.status == 0
+    result = rhel_contenthost.execute('subscription-manager identity')
+    assert result.status == 0
+    # Assert that a yum update was made this day ("Update" or "I, U" in history)
+    timezone_offset = rhel_contenthost.execute('date +"%:z"').stdout.strip()
+    tzinfo = datetime.strptime(timezone_offset, '%z').tzinfo
+    result = rhel_contenthost.execute('yum history | grep U')
```

Comment:
dnf command is not available in rhel7 so this fails on rhel7_contenthost. Since we are using new satellite so we shouldn't worry about getting old history. Wdyt @ogajduse

Reply:
Hmm, good point. I spent some time investigating it and there seem to be no easy way to make it work reasonably on both RHEL7 and RHEL8+.